### PR TITLE
[WIP] basic rgw path trace test with jaeger

### PR DESCRIPTION
**Pull request by suab321321**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -354,6 +354,12 @@ if(WITH_BLKIN)
   include_directories(SYSTEM src/blkin/blkin-lib)
 endif(WITH_BLKIN)
 
+option(WITH_JAEGER "Use jaeger to create tracepoints and display it on JaegerUI" OFF)
```
Comment (ideepika) [inline CMakeLists.txt:357]: for testing can you make you keep it on

**Comment by ideepika** [comment]
@suab321321 I think the build stuff is not the same as in commit:  https://github.com/ceph/ceph/commit/f8fbb94204f140577a532ae17627c8edd27f3f90 and hence we might be seeing make check failures, it's really sensitive to deal with jaeger deps and I think we have to keep our work in sync, it would be ideal if you can integrate your traces with https://github.com/ceph/ceph/commit/f8fbb94204f140577a532ae17627c8edd27f3f90 and test it locally.
Otherwise we are just going down the rabbit hole.
Also, can you keep the trace in an atomic commit and squash all other to it?

**Comment by suab321321** [comment]
@ideepika

**Comment by suab321321** [comment]
@suab321321

**Comment by suab321321** [comment]
@suab321321

**Comment by suab321321** [comment]
@suab321321

**Comment by suab321321** [comment]
@suab321321

**Comment by ideepika** [comment]
@suab321321 can you rebase your PR, sorry for delay getting back to testing your PR

**Comment by suab321321** [comment]
> @suab321321 can you rebase your PR, sorry for delay getting back to testing your PR

no issues I will do it as soon as I can

**Comment by suab321321** [comment]
@ideepika

**Comment by suab321321** [comment]
@ideepika done with rebasing

**Comment by ideepika** [comment]
thanks, testing your PR

**Comment by rzarzynski** [comment]
@ideepika, @cbodley: is this worth a resurrection or should we close it?

**Comment by ideepika** [comment]
I think the efforts inspired by this PR thanks to Abhinav has been continued by RGW team in smaller, more testabl chunks, at this point rebasing seems tedious and I would be fine with it closing and acting as ref for future work.

**Comment by cbodley** [comment]
> I think the efforts inspired by this PR thanks to Abhinav has been continued by RGW team in smaller, more testabl chunks, at this point rebasing seems tedious and I would be fine with it closing and acting as ref for future work.

agreed

**Pull request closed without merging**